labrad/protocol.py

```diff
@@ -416,20 +415,34 @@ def require_secure_connection(auth_type):
 
         if isinstance(credential, auth.Password):
             if not credential.username:
-                # send login packet to get password challenge
-                challenge = yield self._sendManagerRequest()
-
-                # send password response
-                m = hashlib.md5()
-                m.update(challenge)
-                if isinstance(credential.password, bytes):
-                    m.update(credential.password)
+                if 'srp' in self.manager_features:
```

Comment:
@kunalq , thank you.

@maffoo , space permitting consider srp -> secure_random_password, or secure_rnd_passwd.